sycl/source/detail/graph_impl.cpp

```diff
@@ -148,6 +148,25 @@ graph_impl::add(const std::shared_ptr<graph_impl> &Impl,
   return this->add(Handler.MCGType, std::move(Handler.MGraphNodeCG), Dep);
 }
 
+std::shared_ptr<node_impl>
+graph_impl::add(const std::vector<sycl::detail::EventImplPtr> Events){
+
+  std::vector<std::shared_ptr<node_impl>> Deps;
+     
+  // Add any nodes specified by event dependencies into the dependency list
+  for (auto Dep : Events) {
+    if (auto NodeImpl = MEventsMap.find(Dep); NodeImpl != MEventsMap.end()) {
+      Deps.push_back(NodeImpl->second);
+    } else {
+      throw sycl::exception(errc::invalid,
```

Comment:
This feedback on the public PR probably appies here too https://github.com/intel/llvm/pull/9728#discussion_r1222081634